### ISSUE TITLE
NumPy 1.23 removed `alen` and `asscalar`

### ISSUE THREAD

**Issue by mattip** [serge-sans-paille/pythran#2006]
I am seeing [failures](https://github.com/pypy/binary-testing/actions/workflows/pythran.yml) to run the pythran test suite when using NumPy 1.23. The [error](https://github.com/pypy/binary-testing/runs/7374933069?check_suite_focus=true#step:7:6983) is
```
pythran.syntax.PythranSyntaxError: identifier 'alen' not found in module 'numpy'
```

NumPy removed [the deprecated `alen` and `asscalar` in 1.23](https://numpy.org/devdocs/release/1.23.0-notes.html#expired-deprecations). Is this "just" to remove them from the pythran tests for [`alen`](https://github.com/serge-sans-paille/pythran/blob/8cd4c205ef42a528a6fe11fd73bdb2f182869a83/pythran/tests/test_numpy_func1.py#L210) and [`asscalar`](https://github.com/serge-sans-paille/pythran/blob/8cd4c205ef42a528a6fe11fd73bdb2f182869a83/pythran/tests/test_numpy_func2.py#L466) or is there deeper surgery needed?

There are [more non-fatal warnings](https://github.com/pypy/binary-testing/runs/7374933127?check_suite_focus=true#step:7:7066) when running the tests on Python3.9.

**Comment by serge-sans-paille** [comment]
Yeah, we should mark those as obsolete in the pythran test suite too, based on the numpy version.

**Comment by mattip** [comment]
Should the mark go to NumPy 1.16 for asscalar (released Jan 2019) and 1.18 (released Dec 2019) for alen, since that is when they were deprecated?

**Comment by mattip** [comment]
[This warning](https://github.com/pypy/binary-testing/runs/7374933127?check_suite_focus=true#step:7:7086) is weird: where is `pythran/tables.py:422830`? There is a similar one for `np.bool`, which is also deprecated.
```
/home/runner/work/binary-testing/binary-testing/pythran/pythran/tables.py:422830: DeprecationWarning: `np.complex` is a deprecated alias for the builtin `complex`. To silence this warning, use `complex` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.complex128` here.
```
